### PR TITLE
external: add a new configmap to store external cluster script

### PR DESCRIPTION
**Pull request by Nikhil-Ladha**
Added a new static configmap to store the external cluster script and ship it as part of the rook bundle.

### CODE REVIEW

**Review by travisn — COMMENTED** [review]
Let's ensure the Rook CI will fail a gen-csv check so when the auto-sync PRs update the branch, we can detect when this needs to be refreshed.

**Comment by Nikhil-Ladha** [comment]
> Let's ensure the Rook CI will fail a gen-csv check so when the auto-sync PRs update the branch, we can detect when this needs to be refreshed.

We already have this [check](https://github.com/red-hat-storage/rook/blob/master/.github/workflows/build.yml#L101) introduced when we created a separate CSV/bundle for rook long back.

**Review by parth-gr — CHANGES_REQUESTED** [review]

File: deploy/examples/external-cluster-script-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (Nikhil-Ladha) [reply deploy/examples/external-cluster-script-configmap.yaml:1]: 1. We are going to remove it from the annotation in the next release.
2. What do you mean by CLI? The UI team will do the UI change, we have respective stories for that.

[the remark this answers is not in the document]
Comment (parth-gr) [reply deploy/examples/external-cluster-script-configmap.yaml:1]: >We are going to remove it from the annotation in the next release.
Ok thanks, So lets track down it somewhere or maybe we can have that removal in master too.

>What do you mean by CLI?
Update the documentation

[the remark this answers is not in the document]
Comment (Nikhil-Ladha) [reply deploy/examples/external-cluster-script-configmap.yaml:1]: We have the stories for all the changes, please take a look once and let me know if we are missing anything.

File: deploy/examples/kustomization.yaml [hunk 2]
```diff
@@ -6,3 +6,4 @@ resources:
   - common.yaml
   - operator-openshift.yaml
   - csi/nfs/rbac.yaml
+  - external-cluster-script-configmap.yaml
```
[the remark this answers is not in the document]
Comment (Nikhil-Ladha) [reply deploy/examples/kustomization.yaml:9]: Yeah, just tested it out. The cm is getting created when the rook bundle is installed on the cluster.

[the remark this answers is not in the document]
Comment (parth-gr) [reply deploy/examples/kustomization.yaml:9]: Please update the result in the PR

**Comment by Nikhil-Ladha** [comment]
@travisn if everything looks good can we merge this PR?

**Pull request merged**